packages/fs-detectors/src/workspaces/get-workspace-package-paths.ts

```diff
@@ -103,6 +106,16 @@ async function getPackageJsonWorkspacePackagePaths({
   return getPackagePaths(packages, fs);
 }
 
+async function getNxWorkspacePackagePaths({
+  fs,
+}: GetPackagePathOptions): Promise<string[]> {
+  const workspaceJsonAsBuffer = await fs.readFile('workspace.json');
+  const { projects } = JSON.parse(workspaceJsonAsBuffer.toString());
+
+  const packages: string[] = Object.values(projects) ?? [];
```

Comment:
```suggestion
  const workspace = JSON.parse(workspaceJsonAsBuffer.toString());
  const packages: string[] = Object.values(workspace?.projects ?? {});
```

This needs a test for when `workspace.json` is not the shape you are expecting

Reply:
The shape I am not expecting is empty projects array. But I covered that within the regex when looking for the workspaces. But I'll add a test specifically in this file

Reply:
Oh wait actually `'42-npm-workspace-with-nx'` this is the test case for the odd shape

Reply:
But ill add one for if `projects` doesn't exist

Reply:
`packages/fs-detectors/test/fixtures/43-nx-json-misshaped/workspace.json` I added this example when no projects exists

Reply:
What about if workspace.json is the following:

```json
{
  "projects": "hello"
}
```

Reply:
Let me add that in

Reply:
`'44-nx-json-string'` added test case